src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/BeregningService.kt

```diff
@@ -55,4 +61,46 @@ class BeregningService(
 
         tilkjentYtelseRepository.save(tilkjentYtelse)
     }
+
+    /**
+     * Denne metoden henter alle relaterte behandlinger på en person.
+     * Per fagsak henter man tilkjent ytelse fra:
+     * 1. Behandling som er til godkjenning
+     * 2. Siste behandling som er iverksatt
+     * 3. Filtrer bort behandlinger der barnet ikke lenger finnes
+     */
+    fun hentRelevanteTilkjentYtelserForBarn(
+        barnAktør: Aktør,
+        fagsakId: Long
+    ): List<TilkjentYtelse> {
+        val andreFagsaker = fagsakService.hentFagsakerPåPerson(barnAktør)
+            .filter { it.id != fagsakId }
+
+        return andreFagsaker.mapNotNull { fagsak ->
+            val behandlingSomErSendtTilGodkjenning = behandlingRepository.finnBehandlingerSentTilGodkjenning(
+                fagsakId = fagsak.id
+            ).singleOrNull()
+
+            if (behandlingSomErSendtTilGodkjenning != null) {
+                behandlingSomErSendtTilGodkjenning
+            } else {
+                val godkjenteBehandlingerSomIkkeErIverksattEnda =
+                    behandlingRepository.finnBehandlingerSomHolderPåÅIverksettes(fagsakId = fagsak.id).singleOrNull()
+                if (godkjenteBehandlingerSomIkkeErIverksattEnda != null) {
+                    godkjenteBehandlingerSomIkkeErIverksattEnda
+                } else {
+                    val iverksatteBehandlinger = behandlingRepository.finnIverksatteBehandlinger(fagsakId = fagsak.id)
+                    BehandlingUtils.hentSisteBehandlingSomErIverksatt(iverksatteBehandlinger)
+                }
+            }
+        }.map {
+            hentTilkjentYtelseForBehandling(behandlingId = it.id)
+        }.filter {
+            personopplysningGrunnlagRepository
+                .findByBehandlingAndAktiv(behandlingId = it.behandling.id)
+                ?.barna?.map { barn -> barn.aktør }
+                ?.contains(barnAktør)
+                ?: false
+        }.map { it }
```

Comment:
```suggestion
        }
```
Den mappen kan slettes med mindre det er noe magi som skjer bak her som jeg ikke har forstått 😛

Reply:
lol, ja den kan nok fjernes 😛